config/routes.rb

```diff
@@ -1,6 +1,11 @@
 Rails.application.routes.draw do
+  get '/users/:author_id/posts', to: 'posts#index'
+  get '/users/:author_id/posts/:id', to: 'posts#show'
+  get '/', to: "users#index"
+  get '/users/:id', to: "users#show"
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
   # root "articles#index"
+  resources :users
```

Comment:
- [ ] Kindly protect your routes, do not expose unnecessary endpoints, According to the requirement, make sure that you added code to handle only the 4 required URLs which are users & posts index and show pages, see example below ⬇️ 

````ruby
 resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
````

Reply:
@prolajumokeoni Thank you, done :heavy_check_mark: